### ISSUE TITLE
review colour scheme for accessibility

### ISSUE THREAD

**Issue by sds100** [keymapperorg/KeyMapper#601]

**Comment by scatterthought** [comment]
The base red/blue/green you've used for the light and dark modes are easy for me to distinguish. I think if they turned into deep shades of red/blue/green, they'd be much easier to identify. Saying that, colorblind people can have different reactions. This website might be worth checking out:

https://davidmathlogic.com/colorblind/#%23D81B60-%231E88E5-%23FFC107-%23004D40

Let me know if I can help!

**Comment by sds100** [comment]
Yeah, I was wondering how I could make a theme more accessible for all/most types of colour blindness. That is an interesting webpage! They recommend blue-red themes which are very similar to the colours in the light theme so the colours are fine like you said. I have selected the 'record trigger' button in the screenshot. Is the issue that you can't tell that it is selected clearly? 
![image](https://user-images.githubusercontent.com/16245954/108636786-76ef6080-747f-11eb-8564-1a9ccfc64f6d.png)

**Comment by scatterthought** [comment]
Yeah, the best way I can explain it is that it goes from being a bright red to a faded, washed-out red, and the same for the green 
or blue on the Add Action button. As a result, when I'm navigating to the Record Trigger button, I usually go all the way to the bottom bar, then back up one time to highlight the button.

The blue bar at the bottom isn't as much of a problem, because when things are highlighted (e.g. the Save button) I can see the contrast between the two blues. When a whole element changes colour slightly against a solid background of white/black, I can only tell if I see the moment it changes...and then I'm never quite sure if I've just selected or deselected the item. Of course, that's not a problem when you're tapping a screen or clicking a mouse.

P.S. Colorblindness sucks.

**Comment by sds100** [comment]
Yeah, I just put those colours into that website and they are similar. On my less contrastey screen it is even harder for me for to tell. I see why you want them darker now. I'll see what I can do...

**Comment by scatterthought** [comment]
Awesome. You rock.

On a semi-related note, have you considered adding a "support me" option in the app, or some other way to show gratitude? I don't really want to create a Ko-Fi account, as I don't have any other need for it.

**Comment by sds100** [comment]
I'll add a PayPal link to the readme. Here it is https://www.paypal.com/donate?hosted_button_id=K9NBSSWJY9TVY. Google and Apple have weird rules about having links to 3rd party payment processors in an app. I don't want to take any risks. 😄 They want their cut. Thank you!

**Comment by sds100** [comment]
Would making the button even more washed out when you select also work?

**Comment by scatterthought** [comment]
That's hard to say without seeing it. I wonder if you could bold the text and/or put a high-contrast outline around the button when it's selected? We're talking about solving this with colour, but maybe we should be aiming for a signicant visual difference so that colour isn't the only way to tell.

I'll definitely shoot you some cash later!

**Comment by sds100** [comment]
Hmm, yeah maybe that is a better solution. I'll see what elegant solutions are available. I would have thought Google would have thought of this situation. Maybe I'm doing something wrong with how I've designed it.

**Comment by scatterthought** [comment]
Actually, I've found the same thing with other apps I've side-loaded to Android TV. I think it's just not a major consideration since most Android interactions are touch-based.

**Comment by sds100** [comment]
Is this any better? The top one is the button when it is selected.

**Option 1**
![image](https://user-images.githubusercontent.com/16245954/108640591-84faac80-7492-11eb-9603-16ea50fed326.png)
![image](https://user-images.githubusercontent.com/16245954/108640596-8a57f700-7492-11eb-9b71-7694aaeeb006.png)

**Option 2**
![image](https://user-images.githubusercontent.com/16245954/108640683-081c0280-7493-11eb-9c60-78d309a2a631.png)
![image](https://user-images.githubusercontent.com/16245954/108640686-0f431080-7493-11eb-9402-c6905bd913a1.png)

**Comment by scatterthought** [comment]
I would say Option 1 is better. But they're both definitely an improvement!

**Comment by sds100** [comment]
Cool. What about this?
![image](https://user-images.githubusercontent.com/16245954/108640752-6812a900-7493-11eb-8a2b-1480444ccc74.png)
![image](https://user-images.githubusercontent.com/16245954/108640760-6ba63000-7493-11eb-8ab0-3b5c94113fdf.png)

**Comment by scatterthought** [comment]
Oh yeah, that also looks good. Are you thinking of doing a light skin? I think that could work well.

**Comment by sds100** [comment]
Option 3 is the easiest for me to implement so is that okay?

**Comment by scatterthought** [comment]
Sounds great. I'm looking forward to trying it out.

**Comment by sds100** [comment]
[keymapper-2.3.0-alpha-debug.zip](https://github.com/sds100/KeyMapper/files/6038104/keymapper-2.3.0-alpha-debug.zip)
@scatterthought try this! 🙂

**Comment by scatterthought** [comment]
Hey, that looks good. The white-to-blue is much more obvious. I think it looks good.

The Add button on the main screen is still red--I didn't think about that before. I don't think the white/blue will work as well there, but you could have it be grey and then turn red when it's selected (or something like that).

On a side note, when I tried to choose an action to assign to the button, I wasn't able to scroll the lists. I was only able to move horizontally across the top bar (e.g. App, App Shortcut, etc.).

**Comment by sds100** [comment]
Can you select the items in the Key Code list when choosing an action? Only some screens under the tabs in my Android TV emulator can't be selected.

**Comment by sds100** [comment]
Do you mind if I defer this issue like the other Android TV issues? These aren't trivial fixes and I'm not happy with any code I can think of writing to fix these issues. The app is currently not designed **at all** to be interacted with a DPAD or viewed on a TV which means trying to adapt it is really difficult. The proper fix is to re-design the app because then these issues wouldn't be an issue but that is **a lot** of work. I should have deferred this issue like all the other Android TV issues instead of attempting to fix them.

**Comment by scatterthought** [comment]
I thought that I was able to access everything on the production version of the app, but it turns out that I can't. It looks like only the longer lists work, because when you use the DPAD it jumps way down the list.

Anyway, there's no need to rush this for my benefit, so you should definitely defer until you're ready to tackle it. As things are, I've got the production version set up the way I want it for now.

Thanks!

**Comment by sds100** [comment]
Cool! Thanks :) I didn't want to put your hopes up and then disappoint.

**Comment by sds100** [comment]
The UI of the app has changed drastically in the last 6 years so I'm closing this
